src/lifecycleEvents.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { AnyJson, Dictionary } from '@salesforce/ts-types';
+import * as Debug from 'debug';
+
+type callback = (data: AnyJson) => Promise<void>;
+
+/**
+ * An asynchronous event listener and emitter that follows the singleton pattern.
+ */
+export class Lifecycle {
+  public static getInstance(): Lifecycle {
+    if (!this.instance) {
+      this.instance = new Lifecycle();
+    }
+    return this.instance;
+  }
+
+  private static instance: Lifecycle;
+  private debug = Debug(`sfdx:${this.constructor.name}`);
+  private listeners: Dictionary<callback[]>;
+
+  private constructor() {
+    this.listeners = {};
+  }
+
+  public removeAllListeners(eventName: string) {
```

Comment:
Add doc comments to all public methods. I know it may look weird saying `Remove all listeners for a given event` when the method clearly communicates that, but I think it looks weird in the generated documentation and autocompletion if there isn't anything.